snuba/datasets/groups.py

```diff
@@ -132,7 +122,9 @@ def __init__(self) -> None:
         storage = JoinedStorage(join_structure)
         super().__init__(
             storages=[storage],
-            storage_selector=GroupsQueryStorageSelector(storage),
+            query_plan_builder=SingleTableQueryPlanBuilder(
+                storage=storage, post_processors=[PrewhereProcessor()],
```

Comment:
Why do we pass a joined storage into a `SingleTableQueryPlanBuilder` here? The name implies to me that it should only be building queries for single tables and not joins.

Reply:
The only reason is that the joined storage disappears in PR #830, then there will be a JoinedPlanner that takes multiple storages to join and no JoinedStorage anymore. 
This intermediate step is needed to split that into a separate PR, I can rename it to SingleStorageQueryPlanBuilder like I did for the previous PR (SingleStorageSelector).